### PR TITLE
HHH-15418  Composite primary key with NativeQuery & alias not working

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-15418

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java [hunk 1]
```diff
@@ -1867,10 +1869,15 @@ public String selectFragment(String alias, String suffix) {
 								FetchTiming fetchTiming = fetchable.getMappedFetchOptions().getTiming();
 								final boolean selectable;
 								if ( fetchable instanceof AttributeMapping ) {
-									final int propertyNumber = ( (AttributeMapping) fetchable ).getStateArrayPosition();
-									final int tableNumber = getSubclassPropertyTableNumber( propertyNumber );
-									selectable = !isSubclassTableSequentialSelect( tableNumber )
-											&& propertySelectable[propertyNumber];
+									if ( fetchParent instanceof EmbeddableFetchImpl && ( (EmbeddableFetchImpl) fetchParent ).getReferencedMappingContainer() instanceof EntityIdentifierMapping ) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java:1872]: How about this instead?

```suggestion
									if ( fetchParent instanceof EmbeddableResultGraphNode && ( (EmbeddableResultGraphNode) fetchParent ).getReferencedMappingContainer() instanceof EntityIdentifierMapping ) {
```

**Review by beikov — APPROVED** [review]
LGTM

File: hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java]: @beikov,  I changed it to `if ( fetchParent instanceof EmbeddableResultGraphNode && ( (EmbeddableResultGraphNode) fetchParent ).getReferencedMappingContainer() == getIdentifierMapping() )`, what do you think?

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java]: Even better :)

**Pull request merged**